internal/elasticsearch/security/user.go

```diff
@@ -115,13 +115,18 @@ func resourceSecurityUserPut(ctx context.Context, d *schema.ResourceData, meta i
 
 	var user models.User
 	user.Username = usernameId
+
 	if v, ok := d.GetOk("password"); ok {
 		password := v.(string)
-		user.Password = &password
+		if d.HasChange("password") {
```

Comment:
Does it make sense to simplify logic a bit, e.g. remove the embedded `if` and move `d.HasChange("password")` to line #119?
Same with the `password_hash`.